### PR TITLE
Use the display name in the Longroom database

### PR DESCRIPTION
**Pull request by tbergmen**
* Use the Longroom stored display name if it exists when rendering the display name in Longroom
* Pass the Longroom stored display name to o-comments if it exists when they view a Longroom post
* Delete the Longroom stored display name once we know the user has a comments account

Also fixes a few bugs around sign up form.

I have tested this with a user that doesn't have a display name in Coral. Gone through registration and making a new post while still not having a display name in Coral.

Also tested those scenarios with a user that has a display name in Coral.

**Note to reviewers:**
It may be easier to review this PR by going through the commits one by one.

### CODE REVIEW

File: lib/middlewares/checkPseudonym.js [hunk 1]
```diff
@@ -15,7 +16,15 @@ module.exports = function (req, res, next) {
...
-					next();
+					return db.displayNames.selectById(req.userData.user_id)
+						.then(users => {
+							if (!users || !users.length) {
+								return next();
+							}
+
+							req.userData.pseudonym = crypto.decrypt(users[0].display_name);
+							next();
+						});
```
Comment (ker-an) [inline lib/middlewares/checkPseudonym.js]: I'm a bit confused here - why does the response return multiple `users`? Aren't we searching the Longroom database using an `user_id` which I would assume is unique?

**Comment by ker-an** [comment]
Also, thanks for breaking up the PR into such manageable commits! It really helped me when reviewing it 😄

File: lib/middlewares/checkPseudonym.js (hunk 1 above)
Comment (tbergmen) [reply lib/middlewares/checkPseudonym.js]: That's a good point. `pg-promise` (the library Longroom uses to connect Postgres db) returns an array for that query. But I will check if there is a way of returning a single object.

Comment (tbergmen) [reply lib/middlewares/checkPseudonym.js]: @ker-an This is now updated in this commit: https://github.com/Financial-Times/alphaville-longroom/pull/60/commits/542e94fa11673ea5269cb00de75c3116d7b9e221

**Pull request merged**